File: src/main/java/com/github/sorusclient/client/bootstrap/transformer/Transformer.java
/*
 * Copyright (c) 2022. Danterus
 * Copyright (c) 2022. Sorus Contributors
 *
 * SPDX-License-Identifier: MPL-2.0
 */

package com.github.sorusclient.client.bootstrap.transformer;

public interface Transformer {
    boolean canTransform(String name);
    byte[] transform(String name, byte[] data);}

File: src/main/kotlin/com/github/sorusclient/client/bootstrap/transformer/Transformer.kt
/*
 * Copyright (c) 2022. Danterus
 * Copyright (c) 2022. Sorus Contributors
 *
 * SPDX-License-Identifier: MPL-2.0
 */

package com.github.sorusclient.client.bootstrap.transformer

interface Transformer {

    fun canTransform(name: String): Boolean
    fun transform(name: String, data: ByteArray): ByteArray

}